src/main/java/software/aws/neptune/gremlin/GremlinConnectionProperties.java

```diff
@@ -1095,12 +1098,23 @@ public void setLoadBalancingStrategy(@NonNull final LoadBalancingStrategy strate
     protected void validateProperties() throws SQLException {
         // If IAMSigV4 is specified, we need the region provided to us.
         if (getAuthScheme() != null && getAuthScheme().equals(AuthScheme.IAMSigV4)) {
-            final String region = System.getenv().get("SERVICE_REGION");
-            if (region == null) {
-                throw new SQLException(
-                        "Error: SERVICE_REGION environment variable must be set to use IAM authentication. " +
-                                "Note, it must be accessible from BI tools if using a BI tool. " +
-                                "Example SERVICE_REGION=us-east-1.");
+            if ("".equals(getRegion())) {
+                if (System.getenv("SERVICE_REGION") == null) {
+                    throw missingConnectionPropertyError(
+                            "A Service Region must be provided to use IAMSigV4 Authentication through " +
```

Comment:
Text messages which are exposed to users should be part of the localized property file, not hardcoded.

Reply:
I think this could be part of a bigger ticket to consolidate all of our hardcoded error messages. Though I could change it for these for now.